### PR TITLE
feat: Create sharing menu

### LINKED ISSUE #255: HackTogether: Share Property to More social media
**Are you accessing the project from the GitHub Codespaces?**

- [x] Yes, I am using GitHub Codespaces.

**Describe the bug**
When you try to share a property, it directs you to X social media platform.

**To Reproduce**
1. Open the Portal
2. Click on any listing
3. Click the share Icon

**Expected behavior**
Show a pop up or bar with multiple platforms to select from. (e.g. Facebook, LinkedIn, WhatsApp, Messenger, Instagram, and Telegram)
User selects the desired social media to share the property to.

**Screenshots**
Listing Page Share Property
image

X Platform Sign in Page
image

### PR DESCRIPTION
**Pull request by john0isaac**
closes #255 
- Import Material Menu Library.
- Replaced sharing button with sharing Menu.
- Added a string parameter to track the clicked button.
- Redefined Sharing Function.
- Included if statements to check the clicked button and configure sharing accordingly.

While working on this I found a new issue the hashtags weren't added correctly to the twitter sharing link.
I fixed this issue now if you try to share on twitter the hashtags will appear.

Adding more platforms now will be much easier as it's only a copy-and-paste process. (see [this](https://dev.to/shahednasser/how-to-easily-add-share-links-for-each-social-media-platform-1l4f))
image

### CODE REVIEW

File: packages/portal/src/app/rentalpage/rentalpage.component.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { CommonModule } from "@angular/common";
 import { Component, OnInit, inject, signal } from "@angular/core";
 import { ActivatedRoute, Navigation, Router } from "@angular/router";
+import {MatMenuModule} from '@angular/material/menu';
```
Comment (anfibiacreativa) [inline packages/portal/src/app/rentalpage/rentalpage.component.ts:4]: To comply with project conventions, please add a space between curly brackets and import names. Thank you!

File: packages/portal/src/app/shared/listing.service.ts [hunk 2]
```diff
@@ -39,16 +39,27 @@ export class ListingService {
   }
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  async share(listing: Listing) {
+  async share(platform: string, listing: Listing) {
     const rent = listing.fees.at(0);
     const currency = listing.fees.at(-1);
-    this.windowService
+    if(platform === "facebook"){
```
Comment (anfibiacreativa) [inline packages/portal/src/app/shared/listing.service.ts:45]: If we wanted to conform with conventions, there should be a space between the `if` clause and the statement parenthesis. 
We may want here to test if platform is defined at all 

`if (platform && platform === `

But I think we could consider assigning the url value to a conditional value called for example shareUrl. The way I would do it is I would create a global object that has the twitter and facebook urls as properties (so you can extend to more should you need) . 

```
const socialMediaUrls = {
    twitter = '[url-value]',
    facebook = '[url-value]
 ;
```

Then you can import and deconstruct, and execute the conditional for the url value only. // (Sort of quick pseudo code warning here, but you'll get the idea ;) )
```
const { twitter, facebook } = socialMediaUrls;
const shareUrl = (platform) => { platform === 'twitter' ? twitter : facebook; }
```

Then we'd only need to pass shareUrl to the same `this.windowService.native().open(` function.
I see you're also passing the url further down the line and the param name is &u= in one case and &url= in the other. We could also assign that segment of the url to a variable and do the same kind of conditional test to determine what's the corresponding value.

**Review by anfibiacreativa — CHANGES_REQUESTED** [review]

File: packages/portal/src/app/rentalpage/rentalpage.component.ts (hunk 1 above)
Comment (john0isaac) [reply packages/portal/src/app/rentalpage/rentalpage.component.ts]: @anfibiacreativa done, thank you for reviewing!

File: packages/portal/src/app/shared/listing.service.ts (hunk 2 above)
Comment (john0isaac) [reply packages/portal/src/app/shared/listing.service.ts]: @anfibiacreativa Thank you for your amazing feedback I learned a lot!

**Comment by john0isaac** [comment]
- Added Spaces around imports, if statement.
- Added check for the platform value definition.
- Destructure the code for sharing URL, URL of the page.

I couldn't figure out the part where you mentioned creating a **global object** for the sharing URL I did it inside the function as I didn't know if there was a file to create the global variables in or if should I just have included them after the imports in the same file.
I did create an object though inside the share function.

As for the part where you recommend using the **ternary operator** I think it will get more complicated as we add more social media platforms so, I wouldn't recommend it although it's a beautiful idea if we only had two platforms.

Thanks again for the amazing feedback.

**Review by sinedied — APPROVED** [review]
Just added a small suggestion, as Twitter has been renamed to X we might as well do so here 😉

**Comment by john0isaac** [comment]
@sinedied I approved your changes.
Thank you for reviewing!
You can merge now.

**Review by anfibiacreativa — APPROVED** [review]
Thanks for your contribution!

**Review by anfibiacreativa — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**